pkg/authorization/reverse_permission_claims_authorizer_test.go

```diff
@@ -0,0 +1,330 @@
+			wantReason:   "permission claim restricts to \"*\" in API binding name=\"binding\"",
+		},
+		{
+			name: "denied: claimed verb, matching namespace",
+			attr: authorizer.AttributesRecord{Verb: "create", Resource: "configmaps", Namespace: "baz", Name: "bar"},
+			claim: &apisv1alpha1.AcceptablePermissionClaim{
+				PermissionClaim: apisv1alpha1.PermissionClaim{
+					GroupResource: apisv1alpha1.GroupResource{Resource: "configmaps"},
+					ResourceSelector: []apisv1alpha1.ResourceSelector{
+						{Name: "", Namespace: "baz"},
+						{Name: "", Namespace: "buzz"},
+					},
+					Verbs: apisv1alpha1.Verbs{RestrictTo: []string{"get", "list"}},
+				},
+				State: apisv1alpha1.ClaimAccepted,
+			},
+			wantDecision: authorizer.DecisionDeny,
+			wantReason:   "verb is not restricted to in API binding name=\"binding\"",
+		},
+		{
+			name: "denied: claimed verb, all resources claimed",
+			attr: authorizer.AttributesRecord{Verb: "list", Resource: "configmaps", Namespace: "foo", Name: ""},
+			claim: &apisv1alpha1.AcceptablePermissionClaim{
+				PermissionClaim: apisv1alpha1.PermissionClaim{
+					GroupResource: apisv1alpha1.GroupResource{Resource: "configmaps"},
+					All:           true,
+					Verbs:         apisv1alpha1.Verbs{RestrictTo: []string{"get"}},
+				},
+				State: apisv1alpha1.ClaimAccepted,
+			},
+			wantDecision: authorizer.DecisionDeny,
+			wantReason:   "verb is not restricted to in API binding name=\"binding\"",
+		},
+		{
+			name: "denied: claimed verb, namespace claimed",
+			attr: authorizer.AttributesRecord{Verb: "list", Resource: "configmaps", Namespace: "foo", Name: ""},
+			claim: &apisv1alpha1.AcceptablePermissionClaim{
+				PermissionClaim: apisv1alpha1.PermissionClaim{
+					GroupResource: apisv1alpha1.GroupResource{Resource: "configmaps"},
+					Verbs:         apisv1alpha1.Verbs{RestrictTo: []string{"get"}},
+					ResourceSelector: []apisv1alpha1.ResourceSelector{
+						{Name: "", Namespace: "foo"},
+					},
+				},
+				State: apisv1alpha1.ClaimAccepted,
+			},
+			wantDecision: authorizer.DecisionDeny,
+			wantReason:   "verb is not restricted to in API binding name=\"binding\"",
+		},
+		{
+			name: "denied: list (no name and namespace requested) is claimed as some resource selectors are claimed",
+			attr: authorizer.AttributesRecord{Verb: "list", Resource: "configmaps", Namespace: "", Name: ""},
+			claim: &apisv1alpha1.AcceptablePermissionClaim{
+				PermissionClaim: apisv1alpha1.PermissionClaim{
+					GroupResource: apisv1alpha1.GroupResource{Resource: "configmaps"},
+					Verbs:         apisv1alpha1.Verbs{RestrictTo: []string{"get"}},
+					ResourceSelector: []apisv1alpha1.ResourceSelector{
+						{Name: "", Namespace: "foo"},
+					},
+				},
+				State: apisv1alpha1.ClaimAccepted,
+			},
+			wantDecision: authorizer.DecisionDeny,
+			wantReason:   "verb is not restricted to in API binding name=\"binding\"",
+		},
+		{
+			name: "allowed: list (no name and namespace requested) is unclaimed",
```

Comment:
hmm you're right, let me think if this is not a bug actually 🤔

Reply:
ok, as an afterthought this should deny. Strictly, it doesn't match the resource selector but we should in fact deny verbs that cover more than what the resource selector matches so they don't leak.

Reply:
As an afterthought, no, i believe this specific test is not a bug. This permission claim says: "i am restricting you to only list configmaps in namespace `foo`". And that's what we do, we do list configmaps globally, and thus also in namespace `foo`, so we should allow access.

However, if `RestrictTo: []string{}` (empty list) this would translate to "i am restricting you to execute no verb at all on configmaps in namespace `foo`" then this should deny as we list configmaps globally and thus also in namespace `foo`. I added a unit test for this case here.

Finally if `RestrictTo: []string{"create"}` this translates to "i am restricting you to only create configmaps in namespace `foo`" then this should deny as we list (and not create) configmaps globally and thus also in namespace `foo`. I also added a unit test for this case here.